src/slave/containerizer/composing.cpp

```diff
@@ -425,6 +425,28 @@ Future<Containerizer::LaunchResult> ComposingContainerizerProcess::_launch(
 }
 
 
+static Try<Nothing> validateContainerInfoType(const ContainerInfo& info)
```

Comment:
We use this "protobuf union" trick in tons of different protobufs, like the `CheckInfo`, `HealthCheck`, `ExecutorInfo`, `Value`, `Resource::DiskInfo::Source`, `Offer::Operation`, `Environment::Variable`, `Secret`, `Image`, `Volume::Source`, and `ContainerInfo`.

I wonder if it would be helpful to make this a templated helper function in `src/common/protobuf_utils.hpp`.  Something like this (which I have not tried):
```
template <typename Info>
static Try<Nothing> validateProtobufUnion(const Info& info) 
{
  const auto* typeDescriptor = Info::Type_descriptor();
  const auto* infoDescriptor = Info::descriptor();
  ...
}
```
With usage like:
```
CHECK_SOME(validateProtobufUnion<ContainerInfo>(info));
```

Reply:
Making this a templated helper makes sense, I'll do that.

Using this helper for all the other similar protobufs looks like another task of an unknown difficulty and volume;)

Reply:
Converted this into a templated helper.